assets/components/NavBar.js

```diff
@@ -30,50 +44,90 @@ const MenuItems = [
     }
 ]
 
-class NavBar extends Component {
-
-    constructor(){
-        super();
-        this.state = {
-            clicked: false
-        }
-        this.handleClick = this.handleClick.bind(this);
-    }
+function SideDrawer(){
+    const [state, setState] = useState({ right: false });
 
-    handleClick() {
-        this.setState({clicked: !this.state.clicked})
-    }
-
-    
-
-    render(){
-        return(
-            <div className="NavBarItems">
-                <h1 className="navbar-logo">
-                    <a href="/" className="logo"><i className="fas fa-eye eye-logo"></i></a>
-                </h1>
-                <div className="menu-icon" onClick={this.handleClick}>
-                    <i className={this.state.clicked ? "fas fa-times" : "fas fa-bars"}></i>
-                </div>
-                <ul className={this.state.clicked ? "nav-menu active" : "nav-menu"}>
-                    {MenuItems.map((item, index) => {
-                        return(
-                            <li key={index}>
-                                <a className={item.cName} href={item.url}>
-                                    {item.title}
-                                </a>
-                            </li>
-                        )
-                    })}
-                </ul>
-                <div className="profile">
-                    <p>Logged in as:</p>
-                    <p className="loggedInAs">{Cookies.get('username')}</p>
-                </div>
-            </div>
-        )
-    }    
+    const toggleDrawer = (anchor, open) => event => {
+    if (
+        event.type === "keydown" &&
+        (event.key === "Tab" || event.key === "Shift")
+      ) {
+        return;
+      }
+  
+      setState({ [anchor]: open });
+    };
+  
+    const sideDrawerList = anchor => (
+        <div
+            className="list"
+            role="presentation"
+            onClick={toggleDrawer(anchor, false)}
+            onKeyDown={toggleDrawer(anchor, false)}
+        >
+            <List component="nav" className="nav-menu-drawer">
+                {MenuItems.map((item, index) => {
+                    return(
+                        <a href={item.url} key={item.title} className={item.cName + "-drawer"}>
+                            <ListItem button>
+                                <ListItemText primary={item.title} />
+                            </ListItem>
+                        </a>
+                    )
+                })}
+            </List>
+        </div>
+    );
+  
+    return (
```

Comment:
Perhaps it'd be better to split up the 'normal' Navbar and the 'side' navbar into separate files for better readability?